src/libsystemd-network/sd-dhcp-server.c

```diff
@@ -704,13 +707,42 @@ static int get_pool_offset(sd_dhcp_server *server, be32_t requested_ip) {
         return be32toh(requested_ip & ~server->netmask) - server->pool_offset;
 }
 
+static int prepare_new_lease(
+                DHCPLease **ret_lease,
+                be32_t address,
+                const DHCPClientId *client_id,
+                const uint8_t chaddr[static ETH_ALEN],
+                be32_t gateway,
+                usec_t expiration) {
+        _cleanup_free_ DHCPLease *lease = new(DHCPLease, 1);
+        if (!lease)
+                return -ENOMEM;
```

Comment:
Please combine the variable declaration and the allocation. Also, add a blank line before the variable declaration:
```
        usec_t expiration) {

_cleanup_free_ DHCPLease *lease = NULL;

lease = new(DHCPLease, 1);
if (!lease)
        return -ENOMEM;
```